src/foam/lib/json/Outputter.java

```diff
@@ -30,7 +30,7 @@
   protected static ThreadLocal<SimpleDateFormat> sdf = new ThreadLocal<SimpleDateFormat>() {
     @Override
     protected SimpleDateFormat initialValue() {
-      SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+      SimpleDateFormat df = new SimpleDateFormat("+yyyyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
```

Comment:
I'm not too keen on this change because it outputs 2019 as 002019. Can't we make this output 4-digit dates as 4-digit dates only only use 6-digit dates for those beyond 9999, since below 999 9 is the common case?

Reply:
Yes. It will complicate the outputting logic a little but if you want to support dates further in the future or in the past, you'll need to have some conditions in the outputter. The simplest approach would be to ditch date formatting and just use epoch ms as the serialized format.